2020_RTSS_Shuai_Zhao/src/parallel_provider_consumer.rs

```diff
@@ -57,7 +56,7 @@ pub fn get_f_consumers(
                 }
             }
         }
-        f_consumers.insert(providers[p_i].clone(), f_consumer.clone());
+        f_consumers.insert(providers[p_i].drain(..).collect(), f_consumer);
```

Comment:
This is not as fast as it would be with `clone()`.

```suggestion
        f_consumers.insert(providers[p_i].clone(), f_consumer);
```